precompiles/ibc/ibc.go

```diff
@@ -0,0 +1,170 @@
+package ibc
+
+import (
+	"bytes"
+	"embed"
+	"errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	clienttypes "github.com/cosmos/ibc-go/v3/modules/core/02-client/types"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/vm"
+	pcommon "github.com/sei-protocol/sei-chain/precompiles/common"
+	"math/big"
+)
+
+const (
+	TransferMethod = "transfer"
+)
+
+const (
+	IBCAddress = "0x0000000000000000000000000000000000001008"
+)
+
+var _ vm.PrecompiledContract = &Precompile{}
+
+// Embed abi json file to the executable binary. Needed when importing as dependency.
+//
+//go:embed abi.json
+var f embed.FS
+
+func GetABI() abi.ABI {
+	abiBz, err := f.ReadFile("abi.json")
+	if err != nil {
+		panic(err)
+	}
+
+	newAbi, err := abi.JSON(bytes.NewReader(abiBz))
+	if err != nil {
+		panic(err)
+	}
+	return newAbi
+}
+
+type Precompile struct {
+	pcommon.Precompile
+	address        common.Address
+	transferKeeper pcommon.TransferKeeper
+	evmKeeper      pcommon.EVMKeeper
+
+	TransferID []byte
+}
+
+func NewPrecompile(transferKeeper pcommon.TransferKeeper, evmKeeper pcommon.EVMKeeper) (*Precompile, error) {
+	newAbi := GetABI()
+
+	p := &Precompile{
+		Precompile:     pcommon.Precompile{ABI: newAbi},
+		address:        common.HexToAddress(IBCAddress),
+		transferKeeper: transferKeeper,
+		evmKeeper:      evmKeeper,
+	}
+
+	for name, m := range newAbi.Methods {
+		switch name {
+		case TransferMethod:
+			p.TransferID = m.ID
+		}
+	}
+
+	return p, nil
+}
+
+// RequiredGas returns the required bare minimum gas to execute the precompile.
+func (p Precompile) RequiredGas(input []byte) uint64 {
```

Comment:
@Kbhat1 @codchen for IBC transfer, will this have to be hardcoded as for other precompiles? Any suggested value?

Reply:
Is IBC gas consumption pretty constant or can vary from call-to-call quite a bit? If the latter we can report gas consumption dynamically similar to https://github.com/sei-protocol/sei-chain/blob/seiv2/precompiles/wasmd/wasmd.go#L111-L135